### PR TITLE
Add selected_delivery_option in payment step

### PR DESCRIPTION
**Pull request by julienbourdeau**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Add selected_delivery_option in payment step in order to create a full order summary before paying |
| Type? | improvement |
| Category? | FO |
| BC breaks? | no |
| Deprecations? | no |

### CODE REVIEW

File: classes/checkout/CheckoutPaymentStep.php [hunk 1]
```diff
@@ -35,16 +35,22 @@ public function handleRequest(array $requestParams = array())
...
-                    ->getPaymentOptionsForTemplate(),
-                'conditions_to_approve' => $this
-                    ->conditionsToApproveFinder
-                    ->getConditionsToApproveForTemplate(),
-                'selected_payment_option' => $this->selected_payment_option,
-            )
-        );
+        $deliveryOptions = $this->getCheckoutSession()->getDeliveryOptions();
+        $deliveryOptionKey = $this->getCheckoutSession()->getSelectedDeliveryOption();
+        $selected_delivery_option = $deliveryOptions[$deliveryOptionKey];
```
Comment (maximebiloe) [inline classes/checkout/CheckoutPaymentStep.php]: camelCase 😢 ?

File: classes/checkout/CheckoutPaymentStep.php [hunk 2]
```diff
@@ -35,16 +35,22 @@ public function handleRequest(array $requestParams = array())
...
-                'payment_options' => $this
-                    ->paymentOptionsFinder
-                    ->getPaymentOptionsForTemplate(),
-                'conditions_to_approve' => $this
-                    ->conditionsToApproveFinder
-                    ->getConditionsToApproveForTemplate(),
-                'selected_payment_option' => $this->selected_payment_option,
-            )
-        );
+        $delivery_options = $this->getCheckoutSession()->getDeliveryOptions();
```
Comment (maximebiloe) [inline classes/checkout/CheckoutPaymentStep.php]: When I said `camelCase 😢 ?` I meant: can you put them all in camelCase? No more snake_case in PHP 😄

Comment (mickaelandrieu) [inline classes/checkout/CheckoutPaymentStep.php]: ok

Comment (Quetzacoalt91) [inline classes/checkout/CheckoutPaymentStep.php]: Dat 4k picture *_*

**Pull request merged**